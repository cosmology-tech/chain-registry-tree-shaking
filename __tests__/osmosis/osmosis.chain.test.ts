import chain from '@chain-registry/osmosis/chain';
import { Chain } from '@chain-registry/types';

import type { PartialChain } from '../../src';

it('Chain type', () => {
    // just to check types..
    const myChain: Chain = chain;
    expect(myChain.chain_id).toMatchSnapshot();
})

it('Chain simplified snap', () => {
    const myChain: PartialChain = {
        chain_name: chain.chain_name,
        chain_id: chain.chain_id,
        bech32_prefix: chain.bech32_prefix,
        slip44: chain.slip44,
        staking: chain.staking
    }
    expect(myChain).toMatchSnapshot();
})