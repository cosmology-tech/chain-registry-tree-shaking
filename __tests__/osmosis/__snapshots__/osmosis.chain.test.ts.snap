// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Chain simplified snap 1`] = `
{
  "bech32_prefix": "osmo",
  "chain_id": "osmosis-1",
  "chain_name": "osmosis",
  "slip44": 118,
  "staking": {
    "lock_duration": {
      "time": "1209600s",
    },
    "staking_tokens": [
      {
        "denom": "uosmo",
      },
    ],
  },
}
`;

exports[`Chain type 1`] = `"osmosis-1"`;
